### PR TITLE
remove thoth-sourcemanagement from Kebechet

### PR DESCRIPTION
**Pull request by KPostOffice**
## Related Issues and Dependencies

Giving Kebechet some much needed love.  Remove thoth-sourcemanagement and work directly with `ogr` and `git-python`.  Remove legacy code as well as refactoring `config.py` and the base Manager.

## This introduces a breaking change

- [ ] Yes
- [ ] No

## This Pull Request implements

I removed `sm` from manager instances in favour of `ogr.services.base.BaseGitService` which abstracts away Pagure, GitHub, and Gitlab. I also added `self.project` which is an instance of  `ogr.abstract.Project` because Kebechet managers only act on a specific project.  OGR provides high level abstraction for Projects as well as issues and PRs.  The one feature which OGR doesn't provide which was provided by `thoth-sourcemanagement` is adding assignees to issues, so this was implemented directly in the `BaseManager` class.

### CODE REVIEW

File: kebechet/kebechet_runners.py [hunk 1]
```diff
@@ -0,0 +1,171 @@
...
+from kebechet.managers import REGISTERED_MANAGERS
+
+_LOGGER = logging.getLogger("kebechet")
+
+
+def run_webhook(payload: Dict[str, Any]):
+    """Run Kebechet using payload from webhook."""
+    _payload_ = PayloadParser(payload)
+    parsed_payload = _payload_.parsed_data()
+    if (
```
Comment (fridex) [inline kebechet/kebechet_runners.py:38]: What if conditions in this branch are not met?

File: kebechet/kebechet_runners.py [hunk 2]
```diff
@@ -0,0 +1,171 @@
...
+
+
+def run_analysis(analysis_id: str, origin: str, service: str) -> None:
+    """Run result managers (meant to be triggered automatically)."""
+    scheme, _, host, _, slug, _, _ = urllib3.util.parse_url(origin)
+    slug = slug[1:]
+    namespace = slug.split("/")[0]
+    project = slug.split("/")[1]
+
+    service_url = f"{scheme}://{host}"
```
Comment (fridex) [inline kebechet/kebechet_runners.py:83]: I see these repeated. Could these steps form a separate function?

File: kebechet/kebechet_runners.py [hunk 3]
```diff
@@ -0,0 +1,171 @@
...
+                service=ogr_service,
+                service_type=service_type,
+                parsed_payload=parsed_payload,
+                token=token,
+                metadata=metadata,
+            )
+            instance.run(**manager_configuration)
+        except Exception as exc:  # noqa F841
+            _LOGGER.exception(
+                "An error occurred during run of manager %r %r for %r, skipping",
```
Comment (fridex) [inline kebechet/kebechet_runners.py:166]: For future improvements - we should open an issue to the corresponding repo to have this reported to users.

File: kebechet/kebechet_runners.py [hunk 4]
```diff
@@ -0,0 +1,171 @@
...
+        if manager_name not in enabled_managers:
+            _LOGGER.debug(
+                "Skipping manager %r because it is not in the list of enabled managers.",
+                manager_name,
+            )
+            continue
+
+        kebechet_manager = REGISTERED_MANAGERS.get(manager_name)
+        if not kebechet_manager:
+            _LOGGER.error("Unable to find requested manager %r, skipping", manager_name)
```
Comment (fridex) [inline kebechet/kebechet_runners.py:145]: For future improvements - we should open in issue to the corresponding repo to have this reported to users.

**Review by fridex — COMMENTED** [review]
Very nice! Changes look good to me, few minor comments. 👍🏻

File: kebechet/kebechet_runners.py (hunk 4 above)
Comment (KPostOffice) [reply kebechet/kebechet_runners.py:145]: I'll open an issue regarding this.

Comment (KPostOffice) [reply kebechet/kebechet_runners.py:145]: https://github.com/thoth-station/kebechet/issues/726

**Comment by KPostOffice** [comment]
@fridex PTAL

**Review by fridex — APPROVED** [review]
Tremendous job! Very nice! 👍🏻

Not tested, but I believe you 💯

**Pull request merged**